include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_merge_sort.h

```diff
@@ -228,71 +228,425 @@ struct __merge_sort_leaf_submitter<__internal::__optional_kernel_name<_LeafSortN
+        //                 |      |    /  |            |
+        //                 |     n2   /   |            |
+        //                 |      |  /    |            |
+        //                 |      | /     |            |
+        //                 |      |/      |            |
+        // i_elem_local -> +------+-------+            |
+        //                 |     /                     |
+        //                 |    /                      |
+        //                 |   /                       |
+        //                 |  /                        |
+        //                 | /                         |
+        //       i_elem -> +/                          |
+        //                 |                           |
+        //                 |                           |
+        //                 |                           |
+        //                 |                           |
+        //                 |                           |
+        //                 +---------------------------+
+
+        _IndexT i_elem = 0;       // Global diagonal index
+        _IndexT i_elem_local = 0; // Local diagonal index
+        // Offset to the first element in the subrange (i.e. the first element of the first subrange for merge)
+        _IndexT offset = 0;
+        _IndexT n1 = 0; // Size of the first subrange
+        _IndexT n2 = 0; // Size of the second subrange
+
+        WorkDataArea(const std::size_t __n, const std::size_t __n_sorted, const std::size_t __linear_id,
+                     const std::size_t __chunk)
+        {
+            // Calculate global diagonal index
+            i_elem = __linear_id * __chunk;
+
+            // Calculate local diagonal index
+            i_elem_local = i_elem % (__n_sorted * 2);
+
+            // Calculate offset to the first element in the subrange (i.e. the first element of the first subrange for merge)
+            offset = std::min<_IndexT>(i_elem - i_elem_local, __n);
+
+            // Calculate size of the first and the second subranges
+            n1 = std::min<_IndexT>(offset + __n_sorted, __n) - offset;
+            n2 = std::min<_IndexT>(offset + __n_sorted + n1, __n) - (offset + n1);
+        }
+
+        inline bool
+        is_i_elem_local_inside_merge_matrix() const
+        {
+            return i_elem_local < n1 + n2;
+        }
+    };
+
+    template <typename Rng>
+    struct DropViews
+    {
+        using __drop_view_simple_t = oneapi::dpl::__ranges::drop_view_simple<Rng, _IndexT>;
+
+        __drop_view_simple_t rng1;
+        __drop_view_simple_t rng2;
+
+        DropViews(Rng& __rng, const WorkDataArea& __data_area)
+            : rng1(__rng, __data_area.offset), rng2(__rng, __data_area.offset + __data_area.n1)
+        {
+        }
+    };
+
+    std::size_t
+    tune_amount_of_base_diagonals(std::size_t __n_sorted) const
+    {
+        // TODO required to evaluate this value based on available SLM size for each work-group.
```

Comment:
I suppose this comment doesn't make a lot of sense as it is currently written.  
Either we need to mention also an intent to switch to loading data into SLM, or we should rephrase the comment to avoid mentioning SLM.  As written it gives the impression that we are loading data into SLM, and I don't think that is the case.

Reply:
Now we have this comment at two places: here and in the https://github.com/uxlfoundation/oneDPL/blob/a1aaf97d84e91941b7e9c189adcc4731a4d1200d/include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_merge.h#L236. Let's save as is and fix later this functional.